src/interface/settings_game.py

```diff
@@ -0,0 +1,146 @@
+# class settings
+from Tkinter import *
+import ttk
+from read_config_file import ReadConfigFile
+
+class SettingsGame(Frame):
+
+    def __init__(self, parent):
+        """
+        returns the tkinter frame, and will invoke init_ window() function.
+
+        self.configfile -- will have the config file location where the game will read the Game settings
+
+        """
+        main = self.main = Toplevel(parent)
+        self.file_config = '../../config/config.xml'        
+        self.init_window()
+
+    def init_window(self):
+        """
+        shows the objects to be displayed in the window (labels, buttons, listbox, combobox).
+        this function will invoke to the following functions:
+            --fillvaluesintheOutpufileEntry()
+            --ShowListAlgorithms()
+            --showlevelcombobox()
+
+        """
+        self.main.geometry("600x400+200+200")
+        self.main.title("Settings Sudoku Game")
+    
+        #definying the UI option to shown up and edit the values of Output file
+        Label(self.main, text = "Output File Config:").place(x = 15, y= 10)
+        Label(self.main, text = "Path: ").place(x = 15, y = 40)
+        
+        # fill values in the output entry
+        self.fill_values_in_the_output_file_entry()
+        
+        # for future suported types
+        #self.Listofsupportedfilestypes()
+
+        #create a listbox to list all  to list the available algorithms 
+        Label(self.main, text = "Solve Algorithm:").place(x = 15, y = 90)
+        Label(self.main, text = "Algorithm: ").place(x = 15, y = 115)
+        self.show_list_algorithms()
+
+        #Creaate a combo box with levels values
+        Label(self.main, text = " Complexity :").place(x = 15, y= 200)
+        Label(self.main, text = " Level :").place(x = 15, y= 220)
+        
+        #create and fill the combobox
+        self.show_level_combobox()
+
+        Label(self.main, text = "TopLimit: ").place(x = 15, y = 250)
+        Label(self.main, text = "BottonLimit: ").place(x = 15, y = 280)
+        
+        
+        #buttons
+        save_button = Button(self.main, text = " Save ", width = 10, heigh = 2)
+        save_button.place(x = 100, y = 330)
+
+        close_button = Button(self.main, text = " Close ", width = 10, heigh = 2, command = self.quit)
+        close_button.place(x = 200, y = 330)
+
+        self.main.mainloop()
+
+    def fill_values_in_the_output_file_entry(self):
+        """
+        shows the value in output file box.
+        getvaluesoutputfile -- the string value get from "readconfigfile" class to get the Output file settings saved in config file.
+
+        """
+        get_values_output_file = ReadConfigFile(self.file_config).get_output_file()
+        current_path = StringVar()
+        
+        current_path.set(get_values_output_file[0] + '\\' + get_values_output_file[1] + '.' + get_values_output_file[2])
+        
```

Comment:
be consistent with your empty lines, sometimes there are one, two or three lines of separation in the code, and one is more than enough